librna/rnalib.c

```diff
@@ -1073,16 +1074,41 @@ double mk_pf(double x) {
   return exp((-1.0 * x/100.0) / (GASCONST/1000 * (temperature + K0)));
 }
 
+double getScaleValue(int x) {
+  static bool init = false;
+  static const double mean_nrg = -0.1843;
+  static double lookup[10000];
```

Comment:
can we please define a const variable for the threshold of 10000, since it has to be in sync with line 1096 and 1089

Reply:
Unfortunately C doesn't allow the initialization of static arrays with a non-constant size. It doesn't even regard a `const` variable as that and throws a compiler error when using a `const` variable as the array storage size. What I could do is allocate the array on the heap using `malloc` and simpy assign the resulting pointer to a static pointer, which will then serve as the array, like so:
```
static double* lookup;
static const int size = 10000;
if (!init) {
  lookup = malloc(size * sizeof(double);
  ...
}
```

This would work, but obviously means that we would have to allocate the array on the heap, which takes a little longer than a stack allocation. But this is probably negligible.

Reply:
A macro would work as well.